src/rabbit_reader.erl

```diff
@@ -767,6 +767,7 @@ wait_for_channel_termination(N, TimerRef,
                                "error while terminating:~n~p~n",
                         [self(), ConnName, VHost, User#user.username,
                          CS, Channel, Reason]),
+                    rabbit_core_metrics:channel_closed(ChPid),
```

Comment:
@michaelklishin you're right, to delete stats we listen to events. The core metrics only is not enough